### PR TITLE
Tea 53 get teas via axios

### PR DESCRIPTION
**Pull request by deguzmancory**
### In this PR, I'm adding/changing the following:

- Implemented CORS in the backend to allow communication between the frontend running on `localhost:3000` and the backend API on `localhost:8000`
- Implemented Axios for handling API requests to the backend, specifically for the "teas" data.
- Created a new react component `TestFetch` to demonstrate fetching and displaying tea data from the backend.

### Screenshots / Code Snippets:
CORS Configuration in main.go:
```
// Configure CORS
c := cors.New(cors.Options{
    AllowedOrigins: []string{"http://localhost:3000"}, // Set the allowed origins here
    AllowedMethods: []string{"GET", "POST", "PUT", "DELETE", "OPTIONS"}, // Allowed methods
    AllowedHeaders: []string{"Content-Type"}, // Allowed headers
})

// Create the HTTP server with CORS handler
server := &http.Server{
    Addr: ":8000",
    Handler: c.Handler(s), // Use CORS middleware
    ReadHeaderTimeout: 5 * time.Second,
}
```

Axios Implementation (`teaService.tsx`):
```
// teaService.ts
import axios from 'axios';

const BASE_URL = 'http://localhost:8000/teas/';

// Function to fetch all teas
export const fetchTeas = async (published: boolean): Promise<Tea[]> => {
  try {
    const response = await axios.get(`${BASE_URL}${published}`);
    return response.data;
  } catch (error) {
    console.error('Error fetching teas: ', error);
    throw error;
  }
};

```

React Component to display teas (`TestFetch.tsx`):
```
// TestFetch.tsx
import { useState, useEffect } from 'react';
import { fetchTeas } from '../../Axios/teaService';

export default function TestFetch(): JSX.Element {
  const [teas, setTeas] = useState<Tea[]>([]);

  useEffect(() => {
    fetchTeas(true)
      .then((teasData) => {
        setTeas(teasData);
      })
      .catch((error) => {
        console.error(error);
      });
  }, []);

  return (
    <div>
      <h1>Test Fetch</h1>
      <ul>
        {teas.map((tea) => (
          <li key={tea.id}>
            {/* Display tea details */}
          </li>
        ))}
      </ul>
    </div>
  );
}

```

To test this feature:

1. Delete and rebuild your Docker containers.
2. In your Beekeeper (or other DBMS program), create a tea using the following outline:
```
insert into teas (id, name, img_url, description, brew_time, brew_temp, published)
values (
  '1742bc8e-a057-11ee-8c90-0242ac120002',
  'Darjeeling',
  'https://m.media-amazon.com/images/W/MEDIAX_792452-T2/images/I/71jNPvRDaGL.jpg',
  'Darjeeling tea is a type of black tea produced in India',
  '3 to 5 minutes',
  195,
  true
  );
```
3. Navigate to `localhost:3000/test`

From here, you should be able to see the tea displayed.
image
image

[Related Jira Ticket](https://communiteam.atlassian.net/browse/TEA-<53>)

### CODE REVIEW

File: frontend/src/Axios/teaService.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import axios from 'axios';
+
+const BASE_URL = 'http://localhost:8000/teas/';
```
Comment (angelajfisher) [inline frontend/src/Axios/teaService.tsx:3]: Note for later: Replace this with an env variable

File: backend/cmd/main.go [hunk 2]
```diff
@@ -18,60 +8,85 @@ import (
...
-	// locations
+	// Swagger UI endpoint at /docs
 
-	s.Post("/locations/cities", api.CreateCity(dbPool))
+	s.Docs("/docs", swgui.New)
+
+	// Configure CORS
+
+	c := cors.New(cors.Options{
+		AllowedOrigins:             []string{"http://localhost:3000"}, // Set the allowed origins here
```
Comment (angelajfisher) [inline backend/cmd/main.go:60]: Potential note for later: Replace with env var?

**Review by angelajfisher — APPROVED** [review]
LGTM! Don't let us forget to replace the localhost vars with env equivalents later

**Review by BMLx23 — APPROVED** [review]
Looks good.

**Pull request merged**